--- /home/amilie/Desktop/MavenMCP-1.8.9-master-backup/src/main/java/net/minecraft/entity/player/EntityPlayer.java	2025-12-30 22:08:19.807434807 +0000
+++ net/minecraft/entity/player/EntityPlayer.java	2026-01-02 11:51:10.515904316 +0000
@@ -491,6 +491,7 @@
         super.updateEntityActionState();
         this.updateArmSwingProgress();
         this.rotationYawHead = this.rotationYaw;
+        this.rotationPitchHead = this.rotationPitch;
     }
 
     public void onLivingUpdate()
