--- /home/amilie/Desktop/mavenmcp-raw/src/main/java/net/minecraft/client/renderer/OpenGlHelper.java	2021-11-02 23:13:52.586918000 +0000
+++ net/minecraft/client/renderer/OpenGlHelper.java	2025-12-29 19:17:33.150958434 +0000
@@ -26,7 +26,7 @@
 import org.lwjgl.opengl.GL31;
 import org.lwjgl.opengl.GLContext;
 import oshi.SystemInfo;
-import oshi.hardware.Processor;
+import oshi.hardware.CentralProcessor;
 
 public class OpenGlHelper
 {
@@ -330,7 +330,7 @@
 
         try
         {
-            Processor[] aprocessor = (new SystemInfo()).getHardware().getProcessors();
+            CentralProcessor[] aprocessor = new CentralProcessor[]{(new SystemInfo()).getHardware().getProcessor()};
             cpu = String.format("%dx %s", new Object[] {Integer.valueOf(aprocessor.length), aprocessor[0]}).replaceAll("\\s+", " ");
         }
         catch (Throwable var5)
