--- /home/amilie/Desktop/prepped-client-no-actual-ufunction/src/main/java/net/minecraft/client/gui/Gui.java	2021-11-02 23:13:50.610919000 +0000
+++ /home/amilie/Desktop/client/src/main/java/net/minecraft/client/gui/Gui.java	2025-12-30 13:12:02.597938050 +0000
@@ -38,18 +38,19 @@
         drawRect(x, startY + 1, x + 1, endY, color);
     }
 
-    public static void drawRect(int left, int top, int right, int bottom, int color)
+    // Sillicat changed to doubles. Merge patch with Gui.patches.
+    public static void drawRect(double left, double top, double right, double bottom, int color)
     {
         if (left < right)
         {
-            int i = left;
+            double i = left;
             left = right;
             right = i;
         }
 
         if (top < bottom)
         {
-            int j = top;
+            double j = top;
             top = bottom;
             bottom = j;
         }
