--- /home/amilie/Desktop/MavenMCP-1.8.9-master-backup/src/main/java/net/minecraft/client/Minecraft.java	2025-12-30 22:08:19.777017198 +0000
+++ net/minecraft/client/Minecraft.java	2025-12-29 19:35:35.452340068 +0000
@@ -184,6 +184,8 @@
 import org.lwjgl.opengl.OpenGLException;
 import org.lwjgl.opengl.PixelFormat;
 import org.lwjgl.util.glu.GLU;
+import sillicat.Sillicat;
+import sillicat.event.impl.EventKey;
 
 public class Minecraft implements IThreadListener, IPlayerUsage
 {
@@ -486,6 +488,9 @@
         this.checkGLError("Post startup");
         this.ingameGUI = new GuiIngame(this);
 
+        // Start sillicat :p
+        Sillicat.INSTANCE.init();
+
         if (this.serverName != null)
         {
             this.displayGuiScreen(new GuiConnecting(new GuiMainMenu(), this, this.serverName, this.serverPort));
@@ -938,6 +943,9 @@
     {
         try
         {
+            // Kill sillicat :p
+            Sillicat.INSTANCE.shutdown();
+
             this.stream.shutdownStream();
             logger.info("Stopping!");
 
@@ -1793,6 +1801,9 @@
                     }
                     else
                     {
+                        // Sillicat injection for events :p
+                        Sillicat.BUS.post(new EventKey(k));
+
                         if (k == 1)
                         {
                             this.displayInGameMenu();
