--- /home/amilie/Desktop/mavenmcp-raw/src/main/java/net/minecraft/client/resources/model/ModelRotation.java	2021-11-02 23:13:54.366418000 +0000
+++ net/minecraft/client/resources/model/ModelRotation.java	2025-12-29 19:10:01.079736466 +0000
@@ -119,7 +119,7 @@
 
     public javax.vecmath.Matrix4f getMatrix()
     {
-        return Reflector.ForgeHooksClient_getMatrix.exists() ? (javax.vecmath.Matrix4f)Reflector.call(Reflector.ForgeHooksClient_getMatrix, new Object[] {this}): new javax.vecmath.Matrix4f(this.getMatrix4d());
+        return Reflector.ForgeHooksClient_getMatrix.exists() ? (javax.vecmath.Matrix4f)Reflector.call(Reflector.ForgeHooksClient_getMatrix, new Object[] {this}): new javax.vecmath.Matrix4f();
     }
 
     public EnumFacing rotate(EnumFacing p_rotate_1_)
