--- /home/amilie/Desktop/mavenmcp-raw/src/main/java/net/minecraft/client/gui/inventory/GuiChest.java	2021-11-02 23:13:51.334421000 +0000
+++ net/minecraft/client/gui/inventory/GuiChest.java	2025-12-31 11:30:27.367624263 +0000
@@ -4,7 +4,9 @@
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.inventory.ContainerChest;
 import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.Slot;
 import net.minecraft.util.ResourceLocation;
+import sillicat.module.impl.player.ChestStealer;
 
 public class GuiChest extends GuiContainer
 {
@@ -13,9 +15,12 @@
     private IInventory lowerChestInventory;
     private int inventoryRows;
 
+    public static GuiChest INSTANCE;
+
     public GuiChest(IInventory upperInv, IInventory lowerInv)
     {
         super(new ContainerChest(upperInv, lowerInv, Minecraft.getMinecraft().thePlayer));
+        INSTANCE = this;
         this.upperChestInventory = upperInv;
         this.lowerChestInventory = lowerInv;
         this.allowUserInput = false;
@@ -40,4 +45,41 @@
         this.drawTexturedModalRect(i, j, 0, 0, this.xSize, this.inventoryRows * 18 + 17);
         this.drawTexturedModalRect(i, j + this.inventoryRows * 18 + 17, 0, 126, this.xSize, 96);
     }
+
+    // Chest Stealer stuff.
+    public void steal(){
+        new Thread(() ->{
+            try{
+                for(int i = 0; i < GuiChest.this.inventoryRows * 9; i++){
+                    Slot slot = GuiChest.this.inventorySlots.inventorySlots.get(i);
+                    if(slot.getStack() != null){
+                        Thread.sleep((long) ChestStealer.speed.getVal());
+                        GuiChest.this.handleMouseClick(slot, slot.slotNumber, 0, 1);
+                        GuiChest.this.handleMouseClick(slot, slot.slotNumber, 0, 6);
+                    }
+                }
+            } catch(Exception e){
+                e.printStackTrace();
+            }
+        }).start();
+    }
+
+    public void stash(){
+        new Thread(() ->{
+            try{
+                for(int i = GuiChest.INSTANCE.inventoryRows * 9; i < GuiChest.this.inventoryRows * 9 + 44; i++){
+                    Slot slot = GuiChest.this.inventorySlots.inventorySlots.get(i);
+                    if(slot.getStack() != null){
+                        Thread.sleep((long) ChestStealer.speed.getVal());
+                        GuiChest.this.handleMouseClick(slot, slot.slotNumber, 0, 1);
+                        GuiChest.this.handleMouseClick(slot, slot.slotNumber, 0, 6);
+                    }
+                }
+            } catch(Exception e){
+                e.printStackTrace();
+            }
+        }).start();
+    }
+
+
 }
