--- /home/amilie/Desktop/MavenMCP-1.8.9-master-backup/src/main/java/net/minecraft/client/renderer/entity/RendererLivingEntity.java	2025-12-30 22:08:19.792298077 +0000
+++ net/minecraft/client/renderer/entity/RendererLivingEntity.java	2026-01-02 11:54:00.855092194 +0000
@@ -152,7 +152,11 @@
                     f2 = f1 - f;
                 }
 
-                float f7 = entity.prevRotationPitch + (entity.rotationPitch - entity.prevRotationPitch) * partialTicks;
+                float f7;
+                if(entity == renderManager.livingPlayer)
+                    f7 = entity.prevRotationPitchHead + (entity.rotationPitchHead - entity.prevRotationPitchHead) * partialTicks;
+                else
+                    f7 = entity.prevRotationPitch + (entity.rotationPitch - entity.prevRotationPitch) * partialTicks;
                 this.renderLivingAt(entity, x, y, z);
                 float f8 = this.handleRotationFloat(entity, partialTicks);
                 this.rotateCorpse(entity, f8, f, partialTicks);
